packages/Angular/shared/src/lib/shared.service.ts

```diff
@@ -123,17 +123,17 @@ export class SharedService {
     SharedService.RefreshUserNotifications(); // also call this initially when refreshing the dataset...
   }
 
-  public static GeneratePrimaryKeyValueString(pkVals: PrimaryKeyValue[]): string {
+  public static GenerateKeyValuePairString(pkVals: KeyValuePair[]): string {
     return pkVals.map(pk => pk.FieldName + '|' + pk.Value).join('||');
   }
-  public static ParsePrimaryKeys(entity: EntityInfo, routeSegment: string): PrimaryKeyValue[] {
+  public static ParsePrimaryKeys(entity: EntityInfo, routeSegment: string): KeyValuePair[] {
     if (!routeSegment.includes('|')) {
       // If not, return a single element array with a default field name
       return [{ FieldName: entity.PrimaryKey.Name, Value: routeSegment }];
     }
     else {
       const parts = routeSegment.split('||');
-      const pkVals: PrimaryKeyValue[] = [];
+      const pkVals: KeyValuePair[] = [];
```

Comment:
@JS-BC  why not CompositeKey here? MORE IMPORTANTLY - why are you still doing primary key parsing HERE - you should **encapsulate** the parsing of primary keys, system-wide for MJ, into the CompositeKey class so it is only parsed and constructed in one PLACE ever, make sense?